### PR TITLE
Fix deletion guard for Gateway TargetGroupConfigurations still referenced by TCPRoutes

### PR DESCRIPTION
**Pull request by immanuwell**
### Issue

None

### Description

Gateway-targeted `TargetGroupConfiguration`s are a documented path for gateway chaining, but delete reconciliation only guarded Service targets.

So if a `TCPRoute` still points at a `Gateway` backend, deleting the TGC can drop the finalizer and remove the config while it is still referenced. Tiny edge case, but its real.

This patch adds the missing in-use check for `targetReference.kind: Gateway` and covers it with tests.

Repro:
1. Create a `TargetGroupConfiguration` with `targetReference.kind: Gateway`.
2. Create a `TCPRoute` that uses that `Gateway` as a backend.
3. Delete the TGC.
4. Before this patch, the controller can remove the finalizer anyway.
5. With this patch, deletion stays blocked until the `TCPRoute` reference is gone.

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [ ] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by immanuwell** [comment]
@shraddhabang @shuqz please take a look

File: controllers/gateway/targetgroup_configuration_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	elbv2gw "sigs.k8s.io/aws-load-balancer-controller/apis/gateway/v1beta1"
```
Comment (dominikhei) [inline controllers/gateway/targetgroup_configuration_controller_test.go]: Shouldn't this be: sigs.k8s.io/aws-load-balancer-controller/v3/apis/gateway/v1

Comment (immanuwell) [reply controllers/gateway/targetgroup_configuration_controller_test.go]: fixed

**Comment by immanuwell** [comment]
@dominikhei applied your suggestion, please take a look
